fix(routing): partially dynamic segments are being truncated (#10379)
withastro/astro: .changeset/five-colts-rescue.md

---
"astro": patch
---

> Fix the issue of partially truncated dynamic segments.

Reviewer: ```suggestion
Fixes a routing issue with partially truncated dynamic segments.
```
Would be great to add an example as well maybe?
